cargo-dist/templates/installer/installer.sh.j2

```diff
@@ -287,93 +290,84 @@ install() {
-
-        # If we need to override the above
-        if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ] && [ "${CARGO_DIST_FORCE_INSTALL_DIR}" != "$CARGO_HOME" ]; then
-            _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR/bin"
-            _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
-        fi
-
-        # If CARGO_HOME was set but it ended up being the default $HOME-based path,
-        # then keep things late-bound. Otherwise bake the value for safety.
-        # This is what rustup does, and accurately reproducing it is useful.
-        if [ -n "${HOME:-}" ]; then
-            if [ "$HOME/.cargo/bin" = "$_install_dir" ]; then
+    # Before actually consulting the configured install strategy, see
+    # if we're overriding it.
+    if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ]; then
+        {% if install_paths| selectattr("kind", "equalto", "CargoHome") %}
+        _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR/bin"
+        _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
+        _install_dir_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/bin" | sed "s,$HOME,\$HOME,")"
+        _env_script_path_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/env" | sed "s,$HOME,\$HOME,")"
+        {%- else -%}
+        _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR"
+        _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
+        # For these two, if the new value contains $HOME, make
+        # sure to `sed` the $HOME expression back into the expr
+        _install_dir_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR" | sed "s,$HOME,\$HOME,")"
+        _env_script_path_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/env" | sed "s,$HOME,\$HOME,")"
+        {% endif %}
+    # If we're *not* overriding, go ahead with the more complex
+    # path evaluation logic
+    else
+    {% for install_path in install_paths %}
+        if [ -z "${_install_dir:-}" ]; then
+    {% if install_path.kind == "CargoHome" %}
+            # first try CARGO_HOME, then fallback to HOME
+            if [ -n "${CARGO_HOME:-}" ]; then
+                _install_dir="$CARGO_HOME/bin"
+                _env_script_path="$CARGO_HOME/env"
+
+                # If CARGO_HOME was set but it ended up being the default $HOME-based path,
+                # then keep things late-bound. Otherwise bake the value for safety.
+                # This is what rustup does, and accurately reproducing it is useful.
+                if [ -n "${HOME:-}" ]; then
+                    if [ "$HOME/.cargo/bin" = "$_install_dir" ]; then
+                        _install_dir_expr='$HOME/.cargo/bin'
+                        _env_script_path_expr='$HOME/.cargo/env'
+                    else
+                        _install_dir_expr="$_install_dir"
+                        _env_script_path_expr="$_env_script_path"
+                    fi
+                else
+                    _install_dir_expr="$_install_dir"
+                    _env_script_path_expr="$_env_script_path"
+                fi
```

Comment:
Can this all be replaced with the sed approach?

Reply:
Sort of. This is slightly more restricted than the `sed` approach, so any changes we make here would change the behaviour of the installer going forward. My goal here was to minimize side changes like this other than the big one we're changing on purpose.

Reply:
Ah i see what you mean. I at least simplified it a bit.